### ISSUE TITLE
The value string or enum for the property NIC.1.VLanID is of a different type than the property can accept

### ISSUE THREAD

**Issue by Mosibi** [dell/idrac-ansible-module#54]
Running this code

```lang=yaml
- name: Set VLAN tag 1
  local_action: >
     redfish category=Manager command=SetAttributes
     user={{user}} password={{password}} baseuri={{baseuri}}
     mgr_attr_name=NIC.1.VLanID mgr_attr_value=1
  register: set_vlan_tag
```
Results in this error:

```lang=json
fatal: [r640 -> localhost]: FAILED! => {"changed": false, "msg": "Error code {
	u 'error': {
		u 'code': u 'Base.1.0.GeneralError',
		u 'message': u 'A general error has occurred. See ExtendedInfo for more information',
		u '@Message.ExtendedInfo': [{
			u 'Severity': u 'Warning',
			u 'RelatedProperties': [u 'NIC.1.VLanID'],
			u 'MessageId': u 'Base.1.0.PropertyValueTypeError',
			u 'MessageArgs@odata.count': 2,
			u 'RelatedProperties@odata.count': 1,
			u 'Message': u 'The value string or enum for the property NIC.1.VLanID is of a different type than the property can accept.',
			u 'Resolution': u 'Correct the value for the property in the request body and resubmit the request if the operation failed.',
			u 'MessageArgs': [u 'string or enum', u 'NIC.1.VLanID']
		}]
	}
}
```

I suspect the redfisch schema expects a integer instead of a string

**Comment by jose-delarosa** [comment]
@Mosibi, thank for submitting. Let me take a look. This is an attribute I did not test, but if you are correct (needs an int instead of str) then I'll need to convert it somewhere along the way.

**Comment by Mosibi** [comment]
@jose-delarosa 

I tested with a small script and the code below works

```lang=python
#!/bin/python

import urllib3
urllib3.disable_warnings()
import requests
import json

url = 'https://192.168.173.233/redfish/v1/Managers/iDRAC.Embedded.1/Attributes'
payload = {'Attributes': {'NIC.1.VLanID': 1}}
headers = {'content-type': 'application/json'}

response = requests.patch(url, data=json.dumps(payload), headers=headers, verify=False, auth=('root,'calvin'))

print response
```
And setting the vlan id to a string, returns 401

```
payload = {'Attributes': {'NIC.1.VLanID': '1'}}
```

**Comment by jose-delarosa** [comment]
Oh dang I can't test this remotely or I will lose connectivity to my iDRAC. So will update to pass an int, not a string.

**Comment by Mosibi** [comment]
José,

I have a server where i can test, so let me know i can do something for you.

**Comment by jose-delarosa** [comment]
@Mosibi, will work on a patch this week (been busy with other things), and yes if you can test, would appreciate it.

**Comment by jose-delarosa** [comment]
@Mosibi, try attached patch, let me know.
[0242-In-set_manager_attributes-check-if-attribute-value-i.txt](https://github.com/dell/idrac-ansible-module/files/1843840/0242-In-set_manager_attributes-check-if-attribute-value-i.txt)
Thanks.

**Comment by Mosibi** [comment]
Jose,

The patch does not work, concatting the string and the int results in a error:

> An exception occurred during task execution. To see the full traceback, use -vvv. The error was: TypeError: cannot concatenate 'str' and 'int' objects
fatal: [r640 -> localhost]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_FA61jz/ansible_module_redfish.py\", line 291, in <module>\n    main()\n  File \"/tmp/ansible_FA61jz/ansible_module_redfish.py\", line 268, in main\n    result = rf_utils.set_manager_attributes(\"/Attributes\", mgr_attributes)\n  File \"/tmp/ansible_FA61jz/ansible_modlib.zip/ansible/module_utils/redfish_utils.py\", line 615, in set_manager_attributes\nTypeError: cannot concatenate 'str' and 'int' objects\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 0}


This is a patch against the current redfish_utils.py that fix the error message. I tested with this patch and now setting the VLAN works.

```
--- /usr/lib/python2.7/dist-packages/ansible/module_utils/redfish_utils.py.orig	2018-03-26 09:58:54.398709557 +0200
+++ /usr/lib/python2.7/dist-packages/ansible/module_utils/redfish_utils.py	2018-03-26 10:01:00.210780560 +0200
@@ -620,10 +620,10 @@
         # Example: manager_attributes = {\"name\":\"value\"}
         # Check if value is a number. If so, convert to int.
         if attributes['mgr_attr_value'].isdigit():
-           intval = int(attributes['mgr_attr_value'])
-           attributes['mgr_attr_value'] = intval
+            manager_attributes = '{{ "{}": {} }}'.format(attributes['mgr_attr_name'], int(attributes['mgr_attr_value']))
+        else:
+            manager_attributes = '{{ "{}": "{}" }}'.format(attributes['mgr_attr_name'], attributes['mgr_attr_value'])
 
-        manager_attributes = "{\"" + attributes['mgr_attr_name'] + "\":\"" + attributes['mgr_attr_value'] + "\"}"
         payload = {"Attributes": json.loads(manager_attributes) }
         response = self.send_patch_request(self.root_uri + self.manager_uri + uri, payload, HEADERS)
         if response.status_code == 200:

```

**Comment by jose-delarosa** [comment]
@Mosibi thank for the patch. Let me test to make sure we're not introducing regressions, and if OK will ask you to please submit a pull request. Will let you know.

**Comment by jose-delarosa** [comment]
@Mosibi, Please submit a pull request (do a git pull first to refresh your local copy) against the latest master branch so I can merge. Thank you!

**Comment by jose-delarosa** [comment]
Fixed with c3a59678e87e15f48dc204df400a3c19519d681e

**Comment by jose-delarosa** [comment]
Thank you for your contribution.
